blazingcache-core/src/main/java/blazingcache/metrics/Gauge.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Diennea S.r.l. under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Diennea S.r.l. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package blazingcache.metrics;
+
+import blazingcache.utils.RawString;
+
+/**
+ * Simple metric that can be added to or removed from.
+ *
+ * @author dennis.mercuriali
+ */
+public interface Gauge {
+
+    /**
+     * Increment this metric value.
+     * 
+     * @param entryKey key of the entry which triggered this event. Might be {@code null}
+     */
+    void inc(RawString entryKey);
```

Comment:
I have never seen this before in a Metrics system.
why are you doing this ?
because you want to track per-key metrics ?

this way it does not look like a Metrics system but it is more like a callback/hook mechanism

this is quite dangerous because the metrics provider should be as light weight as possible and it must not impact the execution of BC code.

Reply:
We have a multi tenant system where a single cacheclient is used for multiple tenants. To differentiate a cache entry of one tenant from the same entry of another tenant we add a prefix to the actual key.

In blazing cache we end up with something like
tenant1code#entrykey1
tenant2code#entrykey1

We would like to be able to track per-tenant metrics, but this 'tenantcode' extraction should be done by the application, not blazing cache.
With this implementation we can receive the entrykey, extrapolate the code, and log the event on a labeled metric.

Reply:
What about naming this class `GaugeSet` ?

Reply:
If you think it's best, I can revert the last changes (go back to the simple gauge without the RawString param) and create a new issue for implementing a hook mechanism so that we can manage the per-tenant labeled metrics directly in our application.
Still, for our use case, we would need a hook on every operation that is now tracked by the metrics and i fear that it might bloat the code more than the parameterized gauge.

Reply:
IMHO we can keep this change, just renaming `Gauge` to `GaugeSet`